docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,345 @@
+
+```plaintext
+zmk-config/
+├── boards/
+│   ├── shields/
+│   │   ├──sofle/boards/
+│   │   │   └── mikoto_520.overlay
+│   │   └── <other shields>
+│   └── <other boards>
+```
+
+You would then need to make sure that the bus node defined in `mikoto_520.overlay` has the same label as in the other board overlay files of the "sofle".
+
+:::note
+It is _strongly_ recommended that in addition to doing the above as a "temporary" fix, you submit a pull request to ZMK or the external module adding the additional board overlay.
+:::
+
+## Pin Selection
+
+Prior to setting up pin control, you will want to make sure that the pins you use are suitable for the driver that you are using. For this purpose, you will most likely need to consult the datasheet of your MCU/SoC directly. Different MCUs/SoCs have different restrictions in regards to this. The nRF52840 and rp2040 are excellent examples:
+
+- The nRF52840 has its pins marked as either low frequency or high frequency.
+  - Low frequency pins are limited to a frequency of 10KHz, making them unusable for I2C, UART, and (most of) SPI.
+  - High frequency pins can be used for almost all protocols that the nRF52840 is capable of, without restrictions. The exception is QSPI, which is restricted to specific high frequency pins (QSPI is mostly irrelevant for keyboard designs).
+- The rp2040 supports two SPI buses labeled SPI0 and SPI1, two I2C buses labeled I2C0 and I2C1, and two UART buses labeled UART0 and UART1. It restricts its pins to particular functions for each of these. For example, pin 0 can be used as the RX pin of SPI0, the TX pin of UART0, or the SDA pin of I2C0. <br/><br/> You will need to make sure that the pins you select all lie on the same bus and fulfil each of the functions necessary for the bus to function. So for a fully functioning SPI0 bus, you would need to select a SPI0 RX pin, a SPI0 TX pin, and a SPI0 SCK pin.<br/><br/> The rp2040 also comes with programmable input/output blocks (PIO) that can be used to emulate protocols such as (but not limited to) the previously named protocols. Using PIO adds some complexity though, and there are a limited number of PIO blocks, so it is recommended that you stick to the predefined buses where possible.
+
+Note that not all peripherals require all pins to be defined - for example, there are plenty of SPI devices which do not require either the MISO (Main In, Sub Out) or the MOSI (Main Out, Sub In) pin.
+
+## Pin Control File
+
+Pin control consists of two parts:
+
+1. Defining and grouping the pins together for a driver
+2. Defining a driver node and assigning pin groups to it
```

Comment:
"Bus" sounds more familiar to me than "driver", I see you used both below. Maybe add bus here similarly?

One note, for 2. is "overriding/augmenting" rather than "defining" better, since MCU/SoC defines the bus?